client/http/client.go

```diff
@@ -341,520 +248,160 @@ func (c *client) request(
-	var version string
-	err := c.requestWithRetry(ctx,
-		"GetClusterVersion", ClusterVersion,
-		http.MethodGet, nil, &version)
-	if err != nil {
-		return "", err
-	}
-	return version, nil
-}
-
-// GetAllPlacementRuleBundles gets all placement rules bundles.
-func (c *client) GetAllPlacementRuleBundles(ctx context.Context) ([]*GroupBundle, error) {
-	var bundles []*GroupBundle
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRuleBundle", PlacementRuleBundle,
-		http.MethodGet, nil, &bundles)
-	if err != nil {
-		return nil, err
-	}
-	return bundles, nil
-}
-
-// GetPlacementRuleBundleByGroup gets the placement rules bundle by group.
-func (c *client) GetPlacementRuleBundleByGroup(ctx context.Context, group string) (*GroupBundle, error) {
-	var bundle GroupBundle
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRuleBundleByGroup", PlacementRuleBundleByGroup(group),
-		http.MethodGet, nil, &bundle)
-	if err != nil {
-		return nil, err
-	}
-	return &bundle, nil
-}
-
-// GetPlacementRulesByGroup gets the placement rules by group.
-func (c *client) GetPlacementRulesByGroup(ctx context.Context, group string) ([]*Rule, error) {
-	var rules []*Rule
-	err := c.requestWithRetry(ctx,
-		"GetPlacementRulesByGroup", PlacementRulesByGroup(group),
-		http.MethodGet, nil, &rules)
-	if err != nil {
-		return nil, err
-	}
-	return rules, nil
-}
-
-// SetPlacementRule sets the placement rule.
-func (c *client) SetPlacementRule(ctx context.Context, rule *Rule) error {
-	ruleJSON, err := json.Marshal(rule)
-	if err != nil {
-		return errors.Trace(err)
+	for _, member := range membersInfo.Members {
+		if membersInfo.Leader != nil && member.GetMemberId() == membersInfo.Leader.GetMemberId() {
+			newLeaderAddrIdx = len(newPDAddrs)
+		}
+		newPDAddrs = append(newPDAddrs, member.GetClientUrls()...)
 	}
-	return c.requestWithRetry(ctx,
-		"SetPlacementRule", PlacementRule,
-		http.MethodPost, ruleJSON, nil)
-}
-
-// SetPlacementRuleInBatch sets the placement rules in batch.
-func (c *client) SetPlacementRuleInBatch(ctx context.Context, ruleOps []*RuleOp) error {
-	ruleOpsJSON, err := json.Marshal(ruleOps)
-	if err != nil {
-		return errors.Trace(err)
+	// Prevent setting empty addresses.
+	if len(newPDAddrs) == 0 {
+		log.Error("[pd] http client get empty member addresses")
+		return
 	}
-	return c.requestWithRetry(ctx,
-		"SetPlacementRuleInBatch", PlacementRulesInBatch,
-		http.MethodPost, ruleOpsJSON, nil)
-}
-
-// SetPlacementRuleBundles sets the placement rule bundles.
-// If `partial` is false, all old configurations will be over-written and dropped.
-func (c *client) SetPlacementRuleBundles(ctx context.Context, bundles []*GroupBundle, partial bool) error {
-	bundlesJSON, err := json.Marshal(bundles)
-	if err != nil {
-		return errors.Trace(err)
+	oldPDAddrs, oldLeaderAddrIdx := ci.getPDAddrs()
+	ci.setPDAddrs(newPDAddrs, newLeaderAddrIdx)
+	if len(oldPDAddrs) != len(newPDAddrs) || oldLeaderAddrIdx != newLeaderAddrIdx {
```

Comment:
I suggest that we can directly compare the address of the leader and output the address of the leader directly in the log

Reply:
Done. PTAL.